osquery/utils/windows/lzxpress.cpp

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <osquery/logger/logger.h>
+#include <osquery/utils/windows/lzxpress.h>
+
+#include <boost/algorithm/hex.hpp>
+#include <boost/algorithm/string.hpp>
+
+namespace osquery {
+
+std::string decompressLZxpress(std::vector<char> prefetch_data,
+                               unsigned long size) {
+  static HMODULE hDLL =
+      LoadLibraryExW(L"ntdll.dll", NULL, LOAD_LIBRARY_SEARCH_SYSTEM32);
+  if (hDLL == nullptr) {
+    LOG(ERROR) << "Failed to load ntdll.dll";
+    return "Error";
+  }
+
+  typedef HRESULT(WINAPI * pRtlDecompressBufferEx)(
```

Comment:
Prefer to use the `using` syntax instead of `typedef`, and move the types at the top inside an anonymous namespace

Reply:
could u explain this a bit more? Is there an example table implementation that i could reference?
I based this on `windows_security_center.cpp` which uses `typedef`

Reply:
These are C function pointers. So, `using` syntax isn't applicable here.

Reply:
I think @alessandrogario was referring to something like in C++11 syntax:
```
using FP = void(*)(int);
```

Reply:
@farfella thanks for the comments, just to clarify `typedef` is fine for this scenario instead of `using`?

Reply:
I prefer typedef's for C function pointers, but @alessandrogario may have a different opinion here. :)